### ISSUE TITLE
DAGRNN is not deterministic

### ISSUE THREAD

**Issue by bjschoenfeld** [byu-dml/d3m-dynamic-neural-architecture#26]
Comparing two runs with the same random seed, I get the following output. The losses and scores should be identical. This means results are not reproducible.

```
dagrnn_regression regression f1f34085-82eb-4736-a1c7-f54662101143
epoch 0
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 119/119 [00:00<00:00, 175.69it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 119/119 [00:00<00:00, 492.48it/s]
train loss: 0.5581684708595276
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 38/38 [00:00<00:00, 485.24it/s]
validation loss: 2.508641004562378
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 413/413 [00:00<00:00, 595.18it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 128/128 [00:00<00:00, 606.40it/s]
train scores: {'RMSE': 0.5581684301395724}
test scores: {'RMSE': 2.508640777022769}
```

```
dagrnn_regression regression 4212bf85-6f9b-41ff-9855-37fde8a861a4
epoch 0
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 119/119 [00:00<00:00, 172.30it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 119/119 [00:00<00:00, 486.86it/s]
train loss: 0.5988994240760803
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 38/38 [00:00<00:00, 480.56it/s]
validation loss: 2.665893793106079
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 413/413 [00:00<00:00, 604.36it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 128/128 [00:00<00:00, 604.21it/s]
train scores: {'RMSE': 0.5988994168191223}
test scores: {'RMSE': 2.665893801278457}
```

**Comment by erikhuck** [comment]
The DAG RNN is deterministic now. I prefer to not make a pull request until #25 is merged in because there's some code overlap.

**Comment by bjschoenfeld** [comment]
I think it would be better to merge in this bug fix before adding new features like #25. Once this is merged in, then #25 can be updated and fixed as well.
